client/src/components/MultiJurisdictionAudit/AASetup/Participants/index.tsx

```diff
@@ -28,55 +30,70 @@ interface IProps {
   locked: boolean
 }
 
+type IFileSubmitStatus = 'submit' | 'success' | 'failure' | null
```

Comment:
I can see why the changes to this form were a tough challenge, these file uploads have so many states and trying to upload two from one form multiplies the states

One idea I had that might make it simpler: make the `FileForm` component controlled by the `Participants` component

Here's a rough sketch:

```ts
const Participants = (props) => {
  const { electionId } = useParams<{ electionId: string }>()
  const [jurisdictionFile, uploadJurisdictionFile] = useJurisdictionFile(electionId)
  const [standardizedContestsFile, uploadStandardizedContestsFile] = useStandardizedContestsFile(electionId)

  const submit = (values) => {
    // error handling left out for brevity
    await updateSettings({ state: values.state })
    await updateJurisdictionFile(values.jurisdictionFile)
    await updateStandardizedContestsFile(values.standardizedContestsFile)
    // if all the requests succeed, move to the next stage
    if (nextStage.activate) nextStage.activate()
  }

  return (
    <Formik
      initialValues={{
        state: auditSettings.state || '',
        jurisdictionFile: null,
        standardizedContestsFile: null,
      }}
    >
      ... form section for State ...
      <FileForm
        file={jurisdictionFile}
        onFileChange={(file) => setFieldValue("jurisdictionFile", file)}
        ... other props ...
       />
      <FileForm
        file={standardizedContestsFile}
        onFileChange={(file) => setFieldValue("standardizedContestsFile", file)}
        ... other props ...
       />
      // potentially only enable the submit button if both files have been selected
      ... submit button ...
    </Formik>
}
```

Reply:
@jonahkagan That's actually where I went with it initially, but here's the issue I ran into:

We're using the same component for ballot comparison audits and non-ballot-comparison audits, meaning that the standardized contests section may or may not be present at all. This means that the `useStandardizedContests` hook throws a wrench into things, since you cannot ever use a hook conditionally. We can't just say `if (isBallotComparison) useStandardizedContests` for example. We have to conditionally render the standardized contests upload form, and have the hooks related to it managed by that sub component.

So basically, either we duplicate the entire component and have one version include standardized contests uploading and the other not, or we refactor in some fashion similar to how I've got it. The resulting status tracking is... awkward, to say the least, but it's the best I could come up with in order to integrate with the triple form submissions. If you have another idea I'm open to it, though, particularly if there's a bug I haven't found with this setup, haha.

Reply:
Didn't you come up with a solution where we pass a flag into the hook to tell it whether to make the request or not? We could potentially do that with `useStandardizedContests`

Reply:
I was trying to avoid that, but it might actually be the simplest solution here. I'll see what the refactor looks like.